### ISSUE TITLE
SSL client auth

### ISSUE THREAD

**Issue by cshorler** [vi/websocat#111]
A feature request for the client

Here's an extract of my ssh_config

```
Host ssh.example.com
  ProxyCommand websocat -v -E -b --ws-c-uri=wss://%h - ws-c:cmd:'openssl s_client -cert chris.cert.pem -key chris.key.pem -connect %h:443 -quiet'
  User git
  IdentityFile ~/.ssh/id_ed25519
  PubkeyAuthentication yes
```

I didn't find any option to pass client certificate and key without using e.g. `s_client`, although you mention `pkcs12` for listening sockets in the docs.

It would be nice to be able to do something like `curl`
e.g.

```
websocat -E -b --cert chris.cert.pem --key chris.key wss://%h
```

or perhaps a `pkcs12` option for clients, then it's pretty easy to take the same one used by a web browser directly.

**Comment by vi** [comment]
> also, for successful git clone over the above ssh - I have to remove -E otherwise I see I/O error (see below).

Yes, this is expected. You may need to remove `-E` on server side's Websocat instance as well.
Similar problem may arise with HTTP queries over Websocat tunnel.

That's why `-E` is not the default and needs to be specified explicitly.

Unfortunately, stale/closed socket detection may be subpar in Websocat 1, so without `-E` server Websocat instance may leak sockets in some cases (when nobody is trying to send any info to connection clients), that is why `-E` is recommended in CLI message.

**Comment by vi** [comment]
> I wonder if that's what autoreconnect: overlay is for - not sure how to use it with the above line.

I don't expect `autoreconnect:` to be useful for SSH tunneling. Some data may be missed during reconnections and that would break SSH (or TLS) sessions anyway.

`autoreconnect:` is, for example, when you have some skippable status messages that you want to keep receiving even after network was temporarily unavailable. Instead of using Websocat command in a loop, just slap `autoreconnect:` overlay and be done.

**Comment by cshorler** [comment]
good to know these things - thanks!

s_client will work for me in the interim.
